llm/dyn_ext_server.c

```diff
@@ -3,7 +3,7 @@
 #include <stdio.h>
 #include <string.h>
 
-#ifdef __linux__
+#if defined(__linux__) && !defined(__TERMUX__)
```

Comment:
Was it just the `RTLD_DEEPBIND` that wasn't compatible?

Reply:
CPU-only inference works on my Pixel 7 pro (aarch64) using update-to-date Termux (F-Droid) running on top of GrapheneOS (based on AOSP Android 14). I've not attempted to build with the NDK directly but Termux doesn't provide a native GCC compiler (nor do modern NDKs), it uses Clang with GCC compatibility mode. This produces warnings like: _implicit conversion increases floating-point precision_ which I suspect affects newer model quantization formats.

So far, I've had decent success (albeit slow) with the legacy q4 and q5 formats but K_S & K_M not so much. It will be nice if working Vulkan and/or TPU support can eventually be added. Otherwise, without `RTLD_DEEPBIND`, the build succeeds and the dynamic CPU module loads successfully.

Reply:
Given the restructuring we've done recently around moving to fully dynamic loading of the llama.cpp library, I think we can get rid of the deepbind flag and make the linux section more portable.   If you can make that change, I'll test across a handful of systems on my side to confirm no side effects.

Reply:
Done, note that the `else` section is identical to `__linux__`. Builds using my branch for Termux & Archlinux CUDA seem to work as expected.